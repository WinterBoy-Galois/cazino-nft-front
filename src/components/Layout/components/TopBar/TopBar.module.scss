@import '../../../../styles/shared';

.center {
  display: flex;
  align-items: center;
}
.mouse_cursor:hover {
  cursor: pointer;
}
.right {
  justify-content: flex-end;
}

.logo {
  &__size {
    width: 90px;

    @include media-breakpoint-up(lg) {
      width: 128px;
    }
  }

  &__color {
    fill: $ultra-light-blue;
  }
}

.details {
  display: flex;
  align-items: center;
  justify-content: flex-end;

  &__button {
    @include media-breakpoint-up(md) {
      width: 195px;
    }
  }

  &--spacing {
    margin-right: 1rem;
  }
}

.sign-in {
  &__button {
    display: none;

    @include media-breakpoint-up(md) {
      display: unset;
    }
  }

  &__icon {
    width: 36px;
    height: 36px;
    display: block;

    @include media-breakpoint-up(md) {
      display: none;
    }
  }
}
