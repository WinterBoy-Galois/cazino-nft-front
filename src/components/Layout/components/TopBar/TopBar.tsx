import React, { Fragment, useState, useCallback, useRef, useMemo } from 'react';
import { useTranslation } from 'react-i18next';
import styles from './TopBar.module.scss';
import Logo from '../../../icons/Logo';
import SidebarToggle from '../SidebarToggle';
import { useStateValue } from '../../../../state';
import SecondaryButton from '../../../SecondaryButton';
import { ButtonSize } from '../../../Button';
import SignIn from '../../../icons/SignIn';
import Avatar from '../../../Avatar';
import UserMenu from '../../../UserMenu';
// import { Link } from '@reach/router';
import { useBreakpoint } from '../../../../hooks/useBreakpoint.hook';

interface IProps {
  onSignInClick?: () => void;
  onSignOutClick?: () => void;
}

const TopBar: React.FC<IProps> = ({ onSignInClick, onSignOutClick }) => {
  const { t } = useTranslation(['auth']);
  const [{ sidebar, auth }] = useStateValue();
  const [showMenu, setShowMenu] = useState(false);
  const avatarRef = useRef(null);
  const breakpoint = useBreakpoint();

  const handleCloseMenu = useCallback(() => setShowMenu(false), []);
  const handleToggleMenu = useCallback(() => setShowMenu(!showMenu), [showMenu]);

  const isMobileBreakpoint = useMemo(() => {
    switch (breakpoint) {
      case 'xs':
      case 'sm':
        return true;
      default:
        return false;
    }
  }, [breakpoint]);

  const onClickHome = () => {
    window.location.href = '/';
    // props.history.push('/');
  };

  return (
    <Fragment>
      <div className={`container-fluid h-100`}>
        <div className={`row h-100`}>
          <div className={`col-4 ${styles.center}`}>
            {/*<Link to="/">*/}
            {/*  <Logo className={styles.logo__size} fillClassName={styles.logo__color} />*/}
            {/*</Link>*/}
            <div className={styles.mouse_cursor} onClick={onClickHome}>
              <Logo className={styles.logo__size} fillClassName={styles.logo__color} />
            </div>
          </div>
          <div className={`col-8 ${styles.center} ${styles.right}`}>
            <div
              className={`${styles.details} ${
                !sidebar.isOpen || isMobileBreakpoint ? styles['details--spacing'] : ''
              }`}
            >
              {!auth.user ? (
                <Fragment>
                  <div className={styles.details__button}>
                    <SecondaryButton
                      size={ButtonSize.SMALL}
                      className={styles['sign-in__button']}
                      onClick={onSignInClick}
                    >
                      {t('signIn.headline')}
                    </SecondaryButton>
                  </div>
                  <div onClick={onSignInClick}>
                    <SignIn className={styles['sign-in__icon']} />
                  </div>
                </Fragment>
              ) : (
                <Fragment>
                  <span className={styles['details--spacing']}>{auth.user.username}</span>
                  <Avatar
                    avatarUrl={auth.user.avatarUrl}
                    username={auth.user.username}
                    ref={avatarRef}
                    onClick={handleToggleMenu}
                  />
                </Fragment>
              )}
            </div>

            <SidebarToggle arrowLeft={true} show={!sidebar.isOpen} />
          </div>
        </div>
      </div>

      <UserMenu
        show={showMenu}
        username={auth.user?.username}
        onSignOutClick={onSignOutClick}
        onClose={handleCloseMenu}
        toggleRef={avatarRef}
      />
    </Fragment>
  );
};

export default TopBar;
