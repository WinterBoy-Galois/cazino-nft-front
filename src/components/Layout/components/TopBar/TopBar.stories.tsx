import React from 'react';
import { storiesOf } from '@storybook/react';
import TopBar from '.';
import { StateProvider } from '../../../../state';
import { action } from '@storybook/addon-actions';
import { appConfig } from '../../../../common/config';

storiesOf('Components/TopBar', module)
  .add('default', () => <TopBar />)
  .add('with Signed-in user', () => (
    <StateProvider
      state={{
        sidebar: {
          isOpen: false,
          isSound: true,
          selectedTab: 'LATEST_BETS',
          selectedLeaderboardAggregation: 'DAILY',
        },
        modal: {
          type: 'NONE',
        },
        auth: {
          state: 'SIGNED_IN',
          accessToken: 'token',
          user: {
            id: '123',
            isActivated: true,
            username: 'testuser',
            avatarUrl: `${appConfig.apiBasePath}/ava/m1.svg`,
          },
        },
        referral: {},
      }}
    >
      <TopBar onSignOutClick={action('Sign-out')} />
    </StateProvider>
  ));
