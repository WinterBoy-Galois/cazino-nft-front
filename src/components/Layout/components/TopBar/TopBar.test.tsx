import React from 'react';
import { render } from '@testing-library/react';
import TopBar from './TopBar';

describe('TopBar', () => {
  it('should match snapshot', () => {
    // Arrange

    // Act
    const container = render(<TopBar />);

    // Assert
    expect(container).toMatchSnapshot();
  });
});
