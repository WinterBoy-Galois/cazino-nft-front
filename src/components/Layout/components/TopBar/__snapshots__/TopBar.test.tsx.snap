// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container-fluid h-100"
      >
        <div
          class="row h-100"
        >
          <div
            class="col-4 center"
          >
            <a
              aria-current="page"
              href="/"
            >
              <svg
                class="logo__size"
                viewBox="0 0 282.04 138.52"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  data-name="Layer 2"
                  id="Layer_2"
                >
                  <g
                    data-name="Layer 1"
                    id="Layer_1-2"
                  >
                    <path
                      class="logo__color"
                      d="M72.64 86h5.78l5.79-9.4L96.3 57c1.83-3.12 2.16-3.94 2.36-5H76.33c-1.2 0-2.2-.1-3.12-.15a65 65 0 00.48 10.13h1.59s.33-1.87.81-4.08c.34-1.54.87-2.45 1.59-2.78s3.17-.58 13.68-.63L75.28 80.5c-1.78 3.02-2.45 4.5-2.64 5.5zM106.66 51.79h-5.79l-5.78 9.4L83 80.83c-1.82 3.12-2.16 3.94-2.35 5H103c1.2 0 2.21.1 3.12.15a63.8 63.8 0 00-.48-10.13H104s-.34 1.87-.82 4.08c-.33 1.54-.86 2.45-1.58 2.78-.87.44-3.17.58-13.68.63L104 57.31c1.79-3.02 2.46-4.46 2.66-5.52zM109.64 86h5.78l5.79-9.4L133.3 57c1.83-3.12 2.16-3.94 2.36-5h-22.33c-1.2 0-2.2-.1-3.12-.15a65 65 0 00.48 10.13h1.59s.33-1.87.81-4.08c.34-1.54.87-2.45 1.59-2.78s3.17-.58 13.68-.63L112.28 80.5c-1.78 3.02-2.45 4.5-2.64 5.5zM143.66 51.79h-5.79l-5.78 9.4L120 80.83c-1.82 3.12-2.16 3.94-2.35 5H140c1.2 0 2.21.1 3.12.15a63.8 63.8 0 00-.48-10.13H141s-.34 1.87-.82 4.08c-.33 1.54-.86 2.45-1.58 2.78-.87.44-3.17.58-13.68.63L141 57.31c1.79-3.02 2.46-4.46 2.66-5.52zM146.64 86h5.78l5.79-9.4L170.3 57c1.83-3.12 2.16-3.94 2.36-5h-22.33c-1.2 0-2.2-.1-3.12-.15a65 65 0 00.48 10.13h1.59s.33-1.87.81-4.08c.34-1.54.87-2.45 1.59-2.78s3.17-.58 13.68-.63L149.28 80.5c-1.78 3.02-2.45 4.5-2.64 5.5zM180.66 51.79h-5.79l-5.78 9.4L157 80.83c-1.82 3.12-2.16 3.94-2.35 5H177c1.2 0 2.21.1 3.12.15a63.8 63.8 0 00-.48-10.13H178s-.34 1.87-.82 4.08c-.33 1.54-.86 2.45-1.58 2.78-.87.44-3.17.58-13.68.63L178 57.31c1.79-3.02 2.46-4.46 2.66-5.52zM6.58 82.93C2.45 79.86 0 75.06 0 68.39s2.5-10.85 6.58-13.78c3-2.21 7.1-3.31 12.14-3.31a105.74 105.74 0 0111.28 1c.05 3 0 6-.29 9.89h-1.92a31.36 31.36 0 00-.79-4.07c-.77-3-3.89-4.18-8.69-4.18-4.13 0-6.77 1-8.64 2.88-2.5 2.5-3.36 6-3.36 10.75 0 9.17 4.8 15.94 13.54 15.94a18.16 18.16 0 005.61-.77 4.38 4.38 0 003-3c.67-1.68.91-2.5 1.63-4.85h2c-.14 2.73-.53 6.24-1 9a32.78 32.78 0 01-12.43 2.45c-4.35 0-8.82-1.01-12.08-3.41zM33.65 84.23l2.93-.72c.86-.24 1.24-.43 1.53-1.25l9.41-27.41c.86-2.69 1.87-3.31 4.7-4.13a32.35 32.35 0 001.2 3.56l10.81 28a1.43 1.43 0 001.2 1.11l2.78.86v1.63H54v-1.65l3.17-.72c.67-.14 1-.58.72-1.2l-2.5-6.72H43.54l-1.92 6.53c-.24.81-.1 1.2 1.24 1.49l2.93.62v1.63H33.65zm20.64-11.66l-5.52-15s-.53 2.06-1.3 4.65L44.4 72.57zM185.27 84.6l1.39-.29c2.59-.52 2.83-.62 2.88-1.68s.05-3.12.05-5.42V61.37c0-2.31 0-4.32-.05-5.43s-.29-1.15-2.88-1.68l-1.39-.26v-1.75h14.45V54l-1.39.28c-2.6.53-2.84.63-2.88 1.68s0 3.12 0 5.43V82.63c0 1.1.28 1.16 2.88 1.68l1.39.29v1.73h-14.45zM241.38 52.25V54l-1.39.28c-2.59.53-2.83.63-2.88 1.68s-.05 3.12-.05 5.43v25h-3.31l-22-28.08v24.42c0 1.1.29 1.16 2.89 1.68l1.39.29v1.73h-11.42V84.6l1.39-.29c2.6-.52 2.84-.62 2.88-1.68s.05-3.12.05-5.42V61.32c0-2.3 0-4.13-.05-5.23s-.24-1.3-3.5-1.87L204 54v-1.75h10.32l20.07 25.2V61.37v-5.43c0-1.11-.29-1.15-2.88-1.68l-1.4-.28v-1.73zM275.8 82.73A18.69 18.69 0 01263.27 87c-5.81 0-10.66-1.77-13.83-5.61a18.67 18.67 0 01-4-11.76c0-7.06 3.22-12.34 8.21-15.27a20.85 20.85 0 0110.85-2.64 19.91 19.91 0 019.41 2.07c5.23 2.73 8.16 7.87 8.16 14.73a18.67 18.67 0 01-6.27 14.21zm-12-28.37a12.48 12.48 0 00-6.1 1.34c-3.89 2.16-5.9 6.24-5.9 12.63A20.88 20.88 0 00254.53 79a10.57 10.57 0 009 5.24h.47a10.37 10.37 0 009.89-6.44 21.36 21.36 0 001.58-8.49c-.01-8.09-3.7-14.95-11.67-14.95zM182.45 90a14.29 14.29 0 01-27 2.44 14.8 14.8 0 01-.8-2.44H72.14l15.52 35.48a21.71 21.71 0 0028.59 11.19l66.16-28.94A21.69 21.69 0 00195.31 90zM116 125.35a14.3 14.3 0 117.38-18.84 14.3 14.3 0 01-7.38 18.84z"
                    />
                    <circle
                      cx="142.1"
                      cy="26.28"
                      fill="none"
                      r="14.31"
                      transform="matrix(.86 -.5 .5 .86 6.05 75)"
                    />
                    <rect
                      fill="none"
                      height="11"
                      rx="5"
                      width="11"
                      x="187.05"
                      y="33.41"
                    />
                    <path
                      class="logo__color"
                      d="M203.43 28.38l-31.84-7.46a7.32 7.32 0 00-1.6-.21l12 27.7h25.36l2.17-10.24a8.16 8.16 0 00-6.09-9.79zm-5.38 11a5 5 0 01-5 5h-1a5 5 0 01-5-5v-1a5 5 0 015-5h1a5 5 0 015 5zM164.67 13a21.72 21.72 0 00-28.59-11.2L69.92 30.76a22.27 22.27 0 00-12.86 17.65L70 48a14.3 14.3 0 0126.91-2 19.8 19.8 0 01.68 2.36H180l-12-27.7zm-16.84 26.39a14.31 14.31 0 117.38-18.84 14.31 14.31 0 01-7.38 18.84z"
                    />
                  </g>
                </g>
              </svg>
            </a>
          </div>
          <div
            class="col-8 center right"
          >
            <div
              class="details details--spacing"
            >
              <div
                class="details__button"
              >
                <button
                  class="sign-in__button secondary button button--small"
                  type="button"
                >
                  Sign In
                </button>
              </div>
              <div>
                <svg
                  class="sign-in__icon"
                  viewBox="0 0 35.9 38"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    data-name="Layer 2"
                    id="Layer_2"
                  >
                    <g
                      data-name="Слой 5"
                      id="Слой_5"
                    >
                      <path
                        class="cls-1"
                        d="M34.4 27.32a17.5 17.5 0 110-16.66"
                      />
                      <circle
                        cx="19"
                        cy="19"
                        fill="#fff"
                        r="3.07"
                      />
                      <path
                        class="cls-1"
                        d="M24.82 19h9.58M31.57 21.83L34.4 19l-2.83-2.83"
                      />
                    </g>
                  </g>
                </svg>
              </div>
            </div>
            <button
              class="button fade"
              style="transform: rotate(180deg);"
            >
              <svg
                class="icon"
                viewBox="0 0 12 26"
                xmlns="http://www.w3.org/2000/svg"
              >
                <defs>
                  <style>
                    .cls-1{fill:none;stroke:#f4feff;stroke-linecap:round;stroke-width:3px}
                  </style>
                </defs>
                <g
                  data-name="Layer 2"
                  id="Layer_2"
                >
                  <g
                    data-name="Layer 1"
                    id="Layer_1-2"
                  >
                    <g
                      data-name="Group 197"
                      id="Group_197"
                    >
                      <path
                        class="cls-1"
                        d="M1.5 24.5l9-11"
                        data-name="Line 23"
                        id="Line_23"
                      />
                      <path
                        class="cls-1"
                        d="M1.5 1.5l9 12"
                        data-name="Line 24"
                        id="Line_24"
                      />
                    </g>
                  </g>
                </g>
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container-fluid h-100"
    >
      <div
        class="row h-100"
      >
        <div
          class="col-4 center"
        >
          <a
            aria-current="page"
            href="/"
          >
            <svg
              class="logo__size"
              viewBox="0 0 282.04 138.52"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                data-name="Layer 2"
                id="Layer_2"
              >
                <g
                  data-name="Layer 1"
                  id="Layer_1-2"
                >
                  <path
                    class="logo__color"
                    d="M72.64 86h5.78l5.79-9.4L96.3 57c1.83-3.12 2.16-3.94 2.36-5H76.33c-1.2 0-2.2-.1-3.12-.15a65 65 0 00.48 10.13h1.59s.33-1.87.81-4.08c.34-1.54.87-2.45 1.59-2.78s3.17-.58 13.68-.63L75.28 80.5c-1.78 3.02-2.45 4.5-2.64 5.5zM106.66 51.79h-5.79l-5.78 9.4L83 80.83c-1.82 3.12-2.16 3.94-2.35 5H103c1.2 0 2.21.1 3.12.15a63.8 63.8 0 00-.48-10.13H104s-.34 1.87-.82 4.08c-.33 1.54-.86 2.45-1.58 2.78-.87.44-3.17.58-13.68.63L104 57.31c1.79-3.02 2.46-4.46 2.66-5.52zM109.64 86h5.78l5.79-9.4L133.3 57c1.83-3.12 2.16-3.94 2.36-5h-22.33c-1.2 0-2.2-.1-3.12-.15a65 65 0 00.48 10.13h1.59s.33-1.87.81-4.08c.34-1.54.87-2.45 1.59-2.78s3.17-.58 13.68-.63L112.28 80.5c-1.78 3.02-2.45 4.5-2.64 5.5zM143.66 51.79h-5.79l-5.78 9.4L120 80.83c-1.82 3.12-2.16 3.94-2.35 5H140c1.2 0 2.21.1 3.12.15a63.8 63.8 0 00-.48-10.13H141s-.34 1.87-.82 4.08c-.33 1.54-.86 2.45-1.58 2.78-.87.44-3.17.58-13.68.63L141 57.31c1.79-3.02 2.46-4.46 2.66-5.52zM146.64 86h5.78l5.79-9.4L170.3 57c1.83-3.12 2.16-3.94 2.36-5h-22.33c-1.2 0-2.2-.1-3.12-.15a65 65 0 00.48 10.13h1.59s.33-1.87.81-4.08c.34-1.54.87-2.45 1.59-2.78s3.17-.58 13.68-.63L149.28 80.5c-1.78 3.02-2.45 4.5-2.64 5.5zM180.66 51.79h-5.79l-5.78 9.4L157 80.83c-1.82 3.12-2.16 3.94-2.35 5H177c1.2 0 2.21.1 3.12.15a63.8 63.8 0 00-.48-10.13H178s-.34 1.87-.82 4.08c-.33 1.54-.86 2.45-1.58 2.78-.87.44-3.17.58-13.68.63L178 57.31c1.79-3.02 2.46-4.46 2.66-5.52zM6.58 82.93C2.45 79.86 0 75.06 0 68.39s2.5-10.85 6.58-13.78c3-2.21 7.1-3.31 12.14-3.31a105.74 105.74 0 0111.28 1c.05 3 0 6-.29 9.89h-1.92a31.36 31.36 0 00-.79-4.07c-.77-3-3.89-4.18-8.69-4.18-4.13 0-6.77 1-8.64 2.88-2.5 2.5-3.36 6-3.36 10.75 0 9.17 4.8 15.94 13.54 15.94a18.16 18.16 0 005.61-.77 4.38 4.38 0 003-3c.67-1.68.91-2.5 1.63-4.85h2c-.14 2.73-.53 6.24-1 9a32.78 32.78 0 01-12.43 2.45c-4.35 0-8.82-1.01-12.08-3.41zM33.65 84.23l2.93-.72c.86-.24 1.24-.43 1.53-1.25l9.41-27.41c.86-2.69 1.87-3.31 4.7-4.13a32.35 32.35 0 001.2 3.56l10.81 28a1.43 1.43 0 001.2 1.11l2.78.86v1.63H54v-1.65l3.17-.72c.67-.14 1-.58.72-1.2l-2.5-6.72H43.54l-1.92 6.53c-.24.81-.1 1.2 1.24 1.49l2.93.62v1.63H33.65zm20.64-11.66l-5.52-15s-.53 2.06-1.3 4.65L44.4 72.57zM185.27 84.6l1.39-.29c2.59-.52 2.83-.62 2.88-1.68s.05-3.12.05-5.42V61.37c0-2.31 0-4.32-.05-5.43s-.29-1.15-2.88-1.68l-1.39-.26v-1.75h14.45V54l-1.39.28c-2.6.53-2.84.63-2.88 1.68s0 3.12 0 5.43V82.63c0 1.1.28 1.16 2.88 1.68l1.39.29v1.73h-14.45zM241.38 52.25V54l-1.39.28c-2.59.53-2.83.63-2.88 1.68s-.05 3.12-.05 5.43v25h-3.31l-22-28.08v24.42c0 1.1.29 1.16 2.89 1.68l1.39.29v1.73h-11.42V84.6l1.39-.29c2.6-.52 2.84-.62 2.88-1.68s.05-3.12.05-5.42V61.32c0-2.3 0-4.13-.05-5.23s-.24-1.3-3.5-1.87L204 54v-1.75h10.32l20.07 25.2V61.37v-5.43c0-1.11-.29-1.15-2.88-1.68l-1.4-.28v-1.73zM275.8 82.73A18.69 18.69 0 01263.27 87c-5.81 0-10.66-1.77-13.83-5.61a18.67 18.67 0 01-4-11.76c0-7.06 3.22-12.34 8.21-15.27a20.85 20.85 0 0110.85-2.64 19.91 19.91 0 019.41 2.07c5.23 2.73 8.16 7.87 8.16 14.73a18.67 18.67 0 01-6.27 14.21zm-12-28.37a12.48 12.48 0 00-6.1 1.34c-3.89 2.16-5.9 6.24-5.9 12.63A20.88 20.88 0 00254.53 79a10.57 10.57 0 009 5.24h.47a10.37 10.37 0 009.89-6.44 21.36 21.36 0 001.58-8.49c-.01-8.09-3.7-14.95-11.67-14.95zM182.45 90a14.29 14.29 0 01-27 2.44 14.8 14.8 0 01-.8-2.44H72.14l15.52 35.48a21.71 21.71 0 0028.59 11.19l66.16-28.94A21.69 21.69 0 00195.31 90zM116 125.35a14.3 14.3 0 117.38-18.84 14.3 14.3 0 01-7.38 18.84z"
                  />
                  <circle
                    cx="142.1"
                    cy="26.28"
                    fill="none"
                    r="14.31"
                    transform="matrix(.86 -.5 .5 .86 6.05 75)"
                  />
                  <rect
                    fill="none"
                    height="11"
                    rx="5"
                    width="11"
                    x="187.05"
                    y="33.41"
                  />
                  <path
                    class="logo__color"
                    d="M203.43 28.38l-31.84-7.46a7.32 7.32 0 00-1.6-.21l12 27.7h25.36l2.17-10.24a8.16 8.16 0 00-6.09-9.79zm-5.38 11a5 5 0 01-5 5h-1a5 5 0 01-5-5v-1a5 5 0 015-5h1a5 5 0 015 5zM164.67 13a21.72 21.72 0 00-28.59-11.2L69.92 30.76a22.27 22.27 0 00-12.86 17.65L70 48a14.3 14.3 0 0126.91-2 19.8 19.8 0 01.68 2.36H180l-12-27.7zm-16.84 26.39a14.31 14.31 0 117.38-18.84 14.31 14.31 0 01-7.38 18.84z"
                  />
                </g>
              </g>
            </svg>
          </a>
        </div>
        <div
          class="col-8 center right"
        >
          <div
            class="details details--spacing"
          >
            <div
              class="details__button"
            >
              <button
                class="sign-in__button secondary button button--small"
                type="button"
              >
                Sign In
              </button>
            </div>
            <div>
              <svg
                class="sign-in__icon"
                viewBox="0 0 35.9 38"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  data-name="Layer 2"
                  id="Layer_2"
                >
                  <g
                    data-name="Слой 5"
                    id="Слой_5"
                  >
                    <path
                      class="cls-1"
                      d="M34.4 27.32a17.5 17.5 0 110-16.66"
                    />
                    <circle
                      cx="19"
                      cy="19"
                      fill="#fff"
                      r="3.07"
                    />
                    <path
                      class="cls-1"
                      d="M24.82 19h9.58M31.57 21.83L34.4 19l-2.83-2.83"
                    />
                  </g>
                </g>
              </svg>
            </div>
          </div>
          <button
            class="button fade"
            style="transform: rotate(180deg);"
          >
            <svg
              class="icon"
              viewBox="0 0 12 26"
              xmlns="http://www.w3.org/2000/svg"
            >
              <defs>
                <style>
                  .cls-1{fill:none;stroke:#f4feff;stroke-linecap:round;stroke-width:3px}
                </style>
              </defs>
              <g
                data-name="Layer 2"
                id="Layer_2"
              >
                <g
                  data-name="Layer 1"
                  id="Layer_1-2"
                >
                  <g
                    data-name="Group 197"
                    id="Group_197"
                  >
                    <path
                      class="cls-1"
                      d="M1.5 24.5l9-11"
                      data-name="Line 23"
                      id="Line_23"
                    />
                    <path
                      class="cls-1"
                      d="M1.5 1.5l9 12"
                      data-name="Line 24"
                      id="Line_24"
                    />
                  </g>
                </g>
              </g>
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
